payments/payments.go

```diff
@@ -0,0 +1,41 @@
+package payments
+
+import (
+	"context"
+	"net/http"
+)
+
+const (
+	StripeProvider = "stripe"
+	PayPalProvider = "paypal"
+)
+
+// Provider represents a payment provider that can optionally charge, refund,
+// preauthorize payments.
+type Provider interface {
```

Comment:
Right now the required parameters are checked [here](https://github.com/netlify/gocommerce/blob/345a7cd5f13fe5d51ab1107bf389fffd291b75a8/api/payments.go#L94).

If they would be checked upon action, it wouldn't be until [here](https://github.com/netlify/gocommerce/blob/345a7cd5f13fe5d51ab1107bf389fffd291b75a8/api/payments.go#L163) which means an order and transaction are already created. I would also have to check for a specific error to return a 400 instead of a 500.